### ISSUE TITLE
Question: Is it possible to add conditional formatting using a ThreeColorScale in reverse order?

### ISSUE THREAD

**Issue by danstis** [dfinke/ImportExcel#430]
I am looking to add conditional formatting, using a ThreeColorScale but have the low numbers colored green and the high colors in red. Is there any way to achieve this?

Thanks, Dan

**Comment by dfinke** [comment]
Please could you post the script and xlsx, thanks

**Comment by jhoneill** [comment]
Yes. It's possible. You have to use `Add-ConditionalFormat` with the `-PassThru `switch which was new in 5.2 
```
$excel = dir | select name,length,LastWriteTime |export-excel -path "$env:temp\test99.xlsx" -AutoNameRange -PassThru
$rule = Add-ConditionalFormatting -Address $excel.Workbook.Worksheets[1].cells["length"] -RuleType ThreeColorScale  -Passthru
$rule.LowValue.Color = [System.Drawing.Color]::LightGreen
$rule.MiddleValue.Color = [System.Drawing.Color]::LightYellow
$rule.MiddleValue.Type = "Percentile"
$rule.HighValue.Color  = [System.Drawing.Color]::OrangeRed
Close-ExcelPackage $excel -Show
```
By default the middle value is 50 Percent of the maximum but you can change the `.type `(possible values are Formula, Max, Min, Num, Percent, Percentile default is Percent) and `.value` (default is 50) intellisense will complete` [System.Drawing.color]::` for you with known colours

**Comment by dfinke** [comment]
@jhoneill Bummer, I don't see a test for Add-ConditionalFormatting. Did I miss it? I can add a ToDo.

**Comment by jhoneill** [comment]
The tests for add-conditionalformatting  are in export-excel.tests.ps1. I've added some more in my latest commit (PR soon!) .

**Comment by danstis** [comment]
Thanks @jhoneill this explains it a bit better.

Still having issues applying it only to the data section of a table, but that is not directly related to this question, I'll figure that one out.

**Comment by jhoneill** [comment]
I find that I get one of two problems. Either the rule matches the header and formats it, or I format the whole column.  What I have taken to doing is using `-AutoNameRange` to give me a ready made name for each block of cells and I use that - which I had in the example. 
 You can also get the address of the table and specify the cells[  ] part using that. If you get stuck, let me know and I'll try to post and example.

**Comment by jhoneill** [comment]
Just thinking that the Icon sets have a -Reverse switch to display them in reverse order. For the next release I can add put reverse support in for color bars. Then this method would only be needed for custom colors, and swapping the existing ones round would be build in.

**Comment by danstis** [comment]
Thanks, this is what I have currently been running:
`$data = @"
Environment,Name,SKU,Cost
DEV,VM01,Standard_A2_V2,80.03
DEV,DB01,Standard:S4,124.12
DEV,RC01,Standard:C1,84.95
DEV,APP01,Standard:S2 x2,246.24
"@ | ConvertFrom-Csv

# Get test data
$excel = $data | Export-Excel "C:\Temp\Test.xlsx" -WorksheetName "ENVs" -TableStyle "Medium9" -TableName "ENVs" -AutoSize -ClearSheet -PassThru -KillExcel
$excel.Workbook.Worksheets[1].Tables["ENVs"].ShowTotal = $true
$excel.Workbook.Worksheets[1].Tables["ENVs"].Columns["Cost"].TotalsRowFunction = "Sum"
# Add formatting
Set-Format -Address $excel.Workbook.Worksheets[1].Cells["D:D"] -NumberFormat "€ #,##0"
$rule = Add-ConditionalFormatting -Address $excel.Workbook.Worksheets[1].Cells["D:D"] -RuleType ThreeColorScale -Passthru
$rule.LowValue.Color = [System.Drawing.Color]::MediumSeaGreen
$rule.MiddleValue.Color = [System.Drawing.Color]::Khaki
$rule.HighValue.Color  = [System.Drawing.Color]::Tomato
# Write workbook
Close-ExcelPackage $excel -Show
`

And the attached file is the output that I am trying to mirror
[Example.xlsx](https://github.com/dfinke/ImportExcel/files/2355805/Example.xlsx)

I figure it probably needs to be formatted based on a range, before converting to a table.

**Comment by jhoneill** [comment]
I made 3 changes
`Export-Excel -AutoNameRange`
(extra switch. everything else the same)
`Add-ConditionalFormatting -Address $excel.Workbook.Worksheets[1].Cells["Cost"] `
Changed the range to Cost - set by auto name range - everything else the same
```
$rule.MiddleValue.Color = [System.Drawing.Color]::Yellow
$rule.MiddleValue.Type = "Percentile"
```

And that gets the result you have. 


.Net thinks that Standard:C1 is a valid URI so it is getting converted to a hyperlink.
